octavia_f5/api/drivers/f5_driver/tasks/reschedule_tasks.py

```diff
@@ -37,6 +37,30 @@ def __init__(self, **kwargs):
         self._amphora_repo = repo.AmphoraRepository()
 
 
+class LockLoadBalancer(RescheduleTasks):
+    def execute(self, loadbalancer_id):
+        LOG.debug("Locking load balancer: %s ", loadbalancer_id)
+        # Fail if LB already locked. We probably don't want to wait until it's unlocked, since that would mean we're
+        # doing two migrations back-to-back, which we most definitely don't want.
+        if self._amphora_repo.get(db_apis.get_session(), id=loadbalancer_id) \
+                .vrrp_interface == f5_const.RESCHEDULING_LOCK_STRING:
+            return failure.Failure(causes=["Cannot acquire lock for load balancer {}. It's already being rescheduled."])
+        self._amphora_repo.update(db_apis.get_session(), loadbalancer_id,
+                                  vrrp_interface=f5_const.RESCHEDULING_LOCK_STRING)
```

Comment:
This two queries pose a race condition, the status of the loadbalancer could change after the get but before the update. You want this to be done in a transaction. Then you also don't need a revery action since the transaction can be rolled back.

Reply:
I will probably still need a revert function in case a later task fails and the LB needs to be unlocked again.

Reply:
If you do it inside a transaction with rollback, you still don't a revert task.

Reply:
In order to have an automatic transaction rollback **in case a later task fails**, the transaction would have to span those tasks as well, which isn't possible with sqlalchemy as far as I know.

Reply:
You're right!